### DISCUSSION TITLE
Meta-issue: Discordification of Matrix/Synapse/Dendrites (feature request)

### DISCUSSION THREAD

**Discussion by DonaldTsang** [element-hq/element-meta#723] [Ideas]
### Description

There are at least a dozen issues relating to the design of Riot being too user unfriendly, and that people are suggesting designs similar to established apps like Discord and Slack.
(Please post related issues in this page for further discussion)

### Issue Types:

* Better Communities (AKA Discord "servers")
* Seamless E2E with better UX
* Better "unread messages" with ping as primary type
* Tag-based user and mod/admins types (similar to Discord)
* Mumble as a default widget (instead of Jitsi as it is bad on mobile)
* (others not included in the list above)

---

I have been a Riot user for about one and a half years, and I would hope for these features to give people a better user experience (features recommended from friends in Discord)
- [ ] Room feature support
    - [ ] Custom server-based and room-based emojis
       - Have custom server emojis downloaded and be usable as a basic feature
    - [ ] Emoji collection sharing
        - Similar to Telegram, Messenger, LINE and Whatsapp, but shareable and if possible, self-hosted or user-hosted
    - [x] Emoji reaction for texts and files (useful for voting)
    - [x] Automated (non-manual) key sharing
        - Some people are sick of needing to sharing key every time someone gets in the room, they would like to have auto key sharing once in an E2E room
    - [x] Multi-room search
- [ ] Community support
    - [x] Community id generation
    - [ ] Community room mass generation
    - [x] Community room searching
    - [x] Community multi-admin multi-mod support 
    - [ ] Community and sub-community sort
- [ ] Role and access support
    - [ ] Mass room invitation by role or community
        - invitation to a server once application is approved
    - [x] Mass room banning/kicking by role or community
        - for when "unfavorable" behavior by certain users happen
    - [x] Community room sub-categorization and access
        - for visitor vs club member vs club staff separation
    - [ ] Role application and removal
        - for upgrading users to automatically bring them to certain rooms
    - [ ] Announcement rooms by role
        - rooms that allows viewing but not posting unless it is staff
    - [ ] Reaction allowance by role
       - for voting by certain role but not others
- [x] audio chat support
    - [ ] self-host SIP room support (tying SIP with server)
    - [ ] self-host  Mumble room support (tying Mumble with server)
    - [ ] Radio/Playlist bot support
- [ ] collaborative support
    - [ ] self-host Etherpad/Firepad support (tying *pad with server)
    - [ ] self-host Hackmd support (tying Hackmd with server)

Cross-reference: https://github.com/matrix-org/synapse/issues/4030

**Comment by ptman** [comment]
At least I have the option to search multiple rooms in riot

**Comment by ptman** [comment]
Are emoji collections the same thing as sticker packs?

**Comment by ptman** [comment]
Etherpad is available as a widget

**Comment by DonaldTsang** [comment]
@ptman I would say emoji collections are like stickerpack but hosted differently. It should be more fluid, room or server-specific and easy to trade (as a standard file), and also add (if you are a mod/admin)

Still, basic community management is not in Riot Android

**Comment by ptman** [comment]
Reactions: https://github.com/vector-im/riot-web/issues/1357

**Comment by DonaldTsang** [comment]
@ptman that is good progress... but for the other suggestions those features are equally important. Is it possible to get this on spec?

**Comment by ghost** [comment]


**Reply by kittykat** [reply]
Hi, I moved the issue to discussions because it is a collection of cross-platform issues and corresponding discussions that we feel are valuable to retain

**Reply by bew** [reply]
The downside of discussions is that we can't subscribe to the change of state of the issue (receive a notification when it's closed)

**Comment by z411** [comment]
I've seen several clients (Nheko, Cinny, Fluffychat) supporting the adding and usage of server and channel-wide emoji and stickers. If the protocol supports this, isn't just missing in Element only? Or am I misunderstanding?

**Reply by t3chguy** [reply]
@z411 the protocol does not support this, there are multiple approaches that the various clients have taken, including some MSCs, but the spec has not yet accepted any.

**Reply by z411** [reply]
Yes, thank you. After testing them and researching for a bit I noticed this was the case.

**Comment by natrius** [comment]
Just drop this here - https://github.com/johni0702/mumble-web it should be possible to add this as widget to a room already. I haven't tried it, but it might be worth a look. 

I started a discussion about a mumble-widget there https://github.com/mumble-voip/mumble/discussions/6052 if someone might to look into it or add something i missed.

**Comment by heyakyra** [comment]
Possibly related: Filter notifications https://github.com/vector-im/element-meta/issues/2118

**Comment by heyakyra** [comment]
Discord also now supports "forum channels" which is what Zulip paved the way for: https://github.com/element-hq/element-x-android/issues/2831#issuecomment-2722780114

Having a mix of threaded and non-threaded replies in the same room is always such a recipe for chaos. I think each room should either be topic-based threads or without threads at all.

**Reply by erkinalp** [reply]
Forum channels correspond to Matrix's spaces
